test/fuzz/transformation_swap_two_functions_test.cpp

```diff
@@ -0,0 +1,250 @@
+               OpName %48 "value"
+               OpDecorate %48 Location 0
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %6 = OpTypeFloat 32
+          %7 = OpTypePointer Function %6
+          %8 = OpTypeFunction %6 %7
+         %19 = OpConstant %6 2
+         %38 = OpConstant %6 3.70000005
+         %41 = OpConstant %6 3.9000001
+         %44 = OpConstant %6 5
+         %47 = OpTypePointer Input %6
+         %48 = OpVariable %47 Input
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+         %39 = OpVariable %7 Function
+         %42 = OpVariable %7 Function
+         %45 = OpVariable %7 Function
+               OpStore %39 %38
+         %40 = OpFunctionCall %6 %10 %39
+               OpStore %42 %41
+         %43 = OpFunctionCall %6 %13 %42
+               OpStore %45 %44
+         %46 = OpFunctionCall %6 %16 %45
+               OpReturn
+               OpFunctionEnd
+         %10 = OpFunction %6 None %8
+          %9 = OpFunctionParameter %7
+         %11 = OpLabel
+         %18 = OpLoad %6 %9
+         %20 = OpFMul %6 %18 %19
+               OpReturnValue %20
+               OpFunctionEnd
+         %13 = OpFunction %6 None %8
+         %12 = OpFunctionParameter %7
+         %14 = OpLabel
+         %23 = OpVariable %7 Function
+         %24 = OpLoad %6 %12
+               OpStore %23 %24
+         %25 = OpFunctionCall %6 %10 %23
+         %26 = OpFMul %6 %25 %19
+               OpReturnValue %26
+               OpFunctionEnd
+         %16 = OpFunction %6 None %8
+         %15 = OpFunctionParameter %7
+         %17 = OpLabel
+         %29 = OpVariable %7 Function
+         %32 = OpVariable %7 Function
+         %30 = OpLoad %6 %15
+               OpStore %29 %30
+         %31 = OpFunctionCall %6 %10 %29
+         %33 = OpLoad %6 %15
+               OpStore %32 %33
+         %34 = OpFunctionCall %6 %13 %32
+         %35 = OpFMul %6 %31 %34
+               OpReturnValue %35
+               OpFunctionEnd
+  )";
+  const auto env = SPV_ENV_UNIVERSAL_1_3; 
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+
+  // Check context validity.
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // Function should not swap with itself
+
+  
+  // Permutation has invalid values 2
+  ASSERT_DEATH( TransformationSwapTwoFunctions(4,4)
+                   .IsApplicable(context.get(), transformation_context),
+               "Two functions cannot be the same.");
+
+//   // Function with id 29 does not exist. 
+  ASSERT_DEATH( TransformationSwapTwoFunctions(10,29)
+                   .IsApplicable(context.get(), transformation_context),
+               "Function 2 is not in range.");
```

Comment:
Once you have addressed my comments in IsApplicable, this can be changed to ASSERT_FALSE

Reply:
Is there a preference for using ``ASSERT_FALSE`` over ``ASSERT_DEATH`` whenever possible? What are some of the situations where ``ASSERT_DEATH``/``ASSERT_FALSE`` might be preferred

Reply:
ASSERT_FALSE(X, Y) asserts that the expression X evaluates to false. If X does not evaluate to false, the test fails.

ASSERT_DEATH(X, Y) does something completely different. In fact, it is a bit magical. It calls `fork()` to create a child process to evaluate the expression X, and asserts that the process **crashes** (exits with an error status) with a given error message Y. If the error message does not contain Y, then the test fails.

In the above example, I am basically saying that `IsApplicable` should NOT crash when given a function ID that is not in the SPIR-V module. This is a situation that will often happen when using `spirv-fuzz` to shrink a test case by replaying a sublist of transformations. We want to handle this case by returning false so that the transformation is skipped. This is very likely to happen because the shrinker might be given a list of transformations like:

`[ TransformationAddFunction(2, ...), TransformationSwapTwoFunctions(2, 3) ]`

And the shrinker might try to apply a sublist of the transformations:

`[ TransformationSwapTwoFunctions(2, 3) ]`

And in this case, function 2 won't exist, so `TransformationSwapTwoFunctions.IsApplicable()` will return false, and so the transformation won't be applied, which is fine.

Reply:
Fixed.